src/optionsdialog.cpp

```diff
@@ -549,6 +553,53 @@ QString OptionsDialog::readCompilerOptions(QString lang, QString comp)
     return options;
 }
 
+void OptionsDialog::writeTemporaryFileState()
```

Comment:
Please, move writeTemporaryFileState() and readTemporaryFileState() out of OptionsDialog class.

We should stick to single responsibility principle (see SOLID for reference). Class called OptionsDialog - that means that it is dialog that contains UI for Kuzya options. That is its only responsibility. It should not take care of loading temporary files, dealing with main window configuration, etc.  This class should serve only one purpose - manage UI for setting Kuzya options.